attachment_synchronize/models/task.py

```diff
@@ -0,0 +1,210 @@
+            ("move", "Move"),
+            ("move_rename", "Move & Rename"),
+            ("delete", "Delete"),
+        ],
+        help="Action after import a file",
+    )
+    file_type = fields.Selection(
+        selection=[],
+        string="File Type",
+        help="The file type determines an import method to be used "
+        "to parse and transform data before their import in ERP",
+    )
+    enabled = fields.Boolean("Enabled", default=True)
+    check_duplicated_files = fields.Boolean(
+        string="Check duplicated files",
+        help="If checked, will avoid duplication file import",
+    )
+    emails = fields.Char(
+        string="Emails",
+        help="list of email which should be notified in case of failure "
+        "when excuting the files linked to this task",
+    )
+
+    def _prepare_attachment_vals(self, datas, filename):
+        self.ensure_one()
+        vals = {
+            "name": filename,
+            "datas": datas,
+            "datas_fname": filename,
+            "task_id": self.id,
+            "file_type": self.file_type or False,
+        }
+        return vals
+
+    @api.model
+    def _template_render(self, template, record):
+        try:
+            template = mako_template_env.from_string(tools.ustr(template))
+        except Exception:
+            _logger.exception("Failed to load template %r", template)
+
+        variables = {"obj": record}
+        try:
+            render_result = template.render(variables)
+        except Exception:
+            _logger.exception(
+                "Failed to render template %r using values %r"
+                % (template, variables)
+            )
+            render_result = u""
+        if render_result == u"False":
+            render_result = u""
+        return render_result
+
+    @api.model
+    def run_task_import_scheduler(self, domain=None):
+        if domain is None:
+            domain = []
+        domain = expression.AND(
+            [domain, [("method_type", "=", "import"), ("enabled", "=", True)]]
+        )
+        for task in self.search(domain):
+            task.run_import()
+
+    def run_import(self):
+        self.ensure_one()
+        attach_obj = self.env["attachment.queue"]
+        backend = self.backend_id
+        filepath = self.filepath or ""
+        filenames = backend._list(relative_path=filepath, pattern=self.pattern)
+        if self.check_duplicated_files:
+            filenames = self._file_to_import(filenames)
+        total_import = 0
+        for file_name in filenames:
+            with api.Environment.manage():
+                with odoo.registry(self.env.cr.dbname).cursor() as new_cr:
+                    new_env = api.Environment(
+                        new_cr, self.env.uid, self.env.context
+                    )
+                    try:
+                        full_absolute_path = os.path.join(filepath, file_name)
+                        datas = backend._get_b64_data(full_absolute_path)
```

Comment:
```suggestion
                        data = backend._get_b64_data(full_absolute_path)
```
data is already a plural form https://www.learnenglish.de/mistakes/data.html
You should propagate the change...